javaagent-extension-api/src/main/java/io/opentelemetry/javaagent/extension/internal/DeclarativeConfigPropertiesBridgeBuilder.java

```diff
@@ -0,0 +1,105 @@
+ * at any time.
+ */
+public class DeclarativeConfigPropertiesBridgeBuilder {
+  /**
+   * order is important here, so we use LinkedHashMap - see {@link #addMapping(String, String)} for
+   * more details
+   */
+  private final Map<String, String> mappings = new LinkedHashMap<>();
+
+  private final Map<String, Object> overrideValues = new HashMap<>();
+
+  public DeclarativeConfigPropertiesBridgeBuilder() {}
+
+  /**
+   * Adds a mapping from a property prefix to a YAML path.
+   *
+   * <p>For example, if the property prefix is "otel.javaagent" and the YAML path is "agent", then
+   * any property starting with "otel.javaagent." will be resolved against the "agent" node in the
+   * instrumentation/java section of the YAML configuration.
+   *
+   * @param propertyPrefix the prefix of the property to translate
+   * @param yamlPath the YAML path to resolve the property against
+   */
+  @CanIgnoreReturnValue
+  public DeclarativeConfigPropertiesBridgeBuilder addMapping(
+      String propertyPrefix, String yamlPath) {
+    mappings.put(propertyPrefix, yamlPath);
+    return this;
+  }
+
+  /**
+   * Adds a fixed override value for a property.
+   *
+   * @param propertyName the name of the property to override
+   * @param value the value to return when the property is requested
+   */
+  @CanIgnoreReturnValue
+  public DeclarativeConfigPropertiesBridgeBuilder addOverride(String propertyName, Object value) {
+    overrideValues.put(propertyName, value);
+    return this;
+  }
+
+  /** Build {@link ConfigProperties} from the {@code autoConfiguredOpenTelemetrySdk}. */
+  public ConfigProperties build(AutoConfiguredOpenTelemetrySdk autoConfiguredOpenTelemetrySdk) {
+    ConfigProperties sdkConfigProperties =
+        AutoConfigureUtil.getConfig(autoConfiguredOpenTelemetrySdk);
+    if (sdkConfigProperties != null) {
+      return sdkConfigProperties;
+    }
+    ConfigProvider configProvider =
+        AutoConfigureUtil.getConfigProvider(autoConfiguredOpenTelemetrySdk);
+    if (configProvider != null) {
+      return buildFromInstrumentationConfig(configProvider.getInstrumentationConfig());
+    }
+    // Should never happen
+    throw new IllegalStateException(
+        "AutoConfiguredOpenTelemetrySdk does not have ConfigProperties or DeclarativeConfigProperties. This is likely a programming error in opentelemetry-java");
+  }
+
+  /**
+   * Build {@link ConfigProperties} from the {@link DeclarativeConfigProperties} provided by the
+   * instrumentation configuration.
+   *
+   * <p>If the provided {@code instrumentationConfig} is null, an empty {@link
+   * DeclarativeConfigProperties} will be used.
+   *
+   * @param instrumentationConfig the instrumentation configuration to build from
+   * @return a new instance of {@link ConfigProperties}
+   */
+  public ConfigProperties buildFromInstrumentationConfig(
+      @Nullable DeclarativeConfigProperties instrumentationConfig) {
```

Comment:
is null ever passed here?

Reply:
yes, it get's this, which is null if `instrumentation/development` is missing in the config file

```java
public interface ConfigProvider {

  /**
   * Returns the {@link DeclarativeConfigProperties} corresponding to <a
   * href="https://github.com/open-telemetry/opentelemetry-configuration/blob/main/schema/instrumentation.json">instrumentation
   * config</a>, or {@code null} if unavailable.
   *
   * @return the instrumentation {@link DeclarativeConfigProperties}
   */
  @Nullable
  DeclarativeConfigProperties getInstrumentationConfig();
```